File: frontend/src/types/insight.ts
import type { ChartType } from '@/types/dashboard'
import type { DatasetQueryRequest, DatasetQueryResult } from '@/types/dataset'

export interface AskResponse {
  question: string
  answer: string
  query: DatasetQueryRequest
  result: DatasetQueryResult
  // Deterministic (backend rules, not AI-decided) — always a suggestion,
  // never applied without the user being able to change it.
  suggested_chart_type: ChartType
}

export interface Insight {
  id: string
  dataset_id: string
  dataset_filename: string
  question: string
  answer: string
  query_request: DatasetQueryRequest
  row_count: number
  created_at: string
}

File: backend/app/schemas/insight.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.schemas.dataset_query import DatasetQueryRequest


class InsightCreate(BaseModel):
    dataset_id: UUID
    question: str
    answer: str
    query: DatasetQueryRequest
    row_count: int


class InsightRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    dataset_id: UUID
    dataset_filename: str
    question: str
    answer: str
    query_request: DatasetQueryRequest
    row_count: int
    created_at: datetime
